src/test/java/org/objectionary/TokenizerTest.java

```diff
@@ -0,0 +1,80 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.objectionary;
+
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.objectionary.tokens.ArrowToken;
+import org.objectionary.tokens.BracketToken;
+import org.objectionary.tokens.StringToken;
+
+/**
+ * Tokenizer test.
+ *
+ * @since 0.1.0
+ */
+final class TokenizerTest {
+
+    @Disabled
+    @Test
+    void tokenizerTest() {
+        final String input = "ν1(𝜋) ↦ ⟦ 𝜑 ↦ ν2( a ↦ ξ.x ) ⟧";
+        final Tokenizer tokenizer = new Tokenizer(input);
+        MatcherAssert.assertThat(tokenizer.hasNext(), Matchers.equalTo(true));
+        MatcherAssert.assertThat(tokenizer.getToken(), Matchers.instanceOf(StringToken.class));
+        tokenizer.next();
+        MatcherAssert.assertThat(tokenizer.getToken(), Matchers.instanceOf(ArrowToken.class));
+        tokenizer.next();
+        MatcherAssert.assertThat(tokenizer.getToken(), Matchers.instanceOf(BracketToken.class));
+        MatcherAssert.assertThat(
+            (BracketToken) tokenizer.getToken(), Matchers.equalTo(BracketToken.BracketType.OPEN)
+        );
+        tokenizer.next();
+        MatcherAssert.assertThat(tokenizer.getToken(), Matchers.instanceOf(StringToken.class));
+        tokenizer.next();
+        MatcherAssert.assertThat(tokenizer.getToken(), Matchers.instanceOf(ArrowToken.class));
+        tokenizer.next();
+        MatcherAssert.assertThat(tokenizer.getToken(), Matchers.instanceOf(StringToken.class));
+        tokenizer.next();
+        MatcherAssert.assertThat(tokenizer.getToken(), Matchers.instanceOf(StringToken.class));
+        tokenizer.next();
+        MatcherAssert.assertThat(tokenizer.getToken(), Matchers.instanceOf(ArrowToken.class));
+        tokenizer.next();
+        MatcherAssert.assertThat(tokenizer.getToken(), Matchers.instanceOf(StringToken.class));
+        tokenizer.next();
+        MatcherAssert.assertThat(tokenizer.getToken(), Matchers.instanceOf(BracketToken.class));
+        MatcherAssert.assertThat(
+            (BracketToken) tokenizer.getToken(), Matchers.equalTo(BracketToken.BracketType.CLOSE)
+        );
+        tokenizer.next();
+        MatcherAssert.assertThat(tokenizer.getToken(), Matchers.instanceOf(BracketToken.class));
+        MatcherAssert.assertThat(
+            (BracketToken) tokenizer.getToken(), Matchers.equalTo(BracketToken.BracketType.CLOSE)
+        );
+        tokenizer.next();
+        MatcherAssert.assertThat(tokenizer.hasNext(), Matchers.equalTo(false));
```

Comment:
@EugeneDar Maybe this test can be implemented via `while` or `for` loop. What do you think?

Reply:
@Graur sure, I will fix it.